webapp/src/components/GetEmailModal.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+  <form @submit.prevent="submitForm" class="modal-enclosure">
+    <Modal
+      :modelValue="modelValue"
+      @update:modelValue="$emit('update:modelValue', $event)"
+      :closeButton="false"
+      :disableSubmit="Boolean(successMessage) || !Boolean(emailAddress)"
+    >
+      <template v-slot:body>
+        <div class="form-row">
+          <div class="form-group">
+            <label for="sample-id" class="col-form-label"
+              >Enter your email address to receive a sign-in/registration link:</label
+            >
+            <input
+              v-model="emailAddress"
+              type="text"
+              class="form-control"
+              id="email-address"
+              required
+            />
+            <div class="form-error" v-html="emailValidationMessage"></div>
+            <div class="form-success" v-html="successMessage"></div>
+          </div>
+        </div>
+      </template>
+    </Modal>
+  </form>
+</template>
+
+<script>
+import Modal from "@/components/Modal.vue";
+import { requestMagicLink } from "@/server_fetch_utils.js";
+export default {
+  name: "CreateSampleModal",
+  data() {
+    return {
+      emailAddress: null,
+      emailSent: false,
+      emailError: null,
+      emailValidationMessage: "",
+      successMessage: "",
+    };
+  },
+  props: {
+    modelValue: Boolean,
+  },
+  emits: ["update:modelValue"],
+  methods: {
+    async submitForm() {
+      this.validateEmail();
+      if (this.emailValidationMessage) {
+        return;
+      }
+      let response = await requestMagicLink(this.emailAddress);
+      if (response.status == "success") {
+        this.emailSent = true;
+        this.successMessage =
+          "Email sent! Please check your inbox for a link to sign in or register.";
```

Comment:
```suggestion
          "Email sent! Please check your inbox (and spam folder) for a link to sign in or register. It may take a few minutes to arrive.";
```